src/adapters/ai-compliance.ts

```diff
@@ -0,0 +1,191 @@
+      warnings: { type: 'array', items: { type: 'string', maxLength: 200 } },
+      confidence: { type: 'number', minimum: 0, maximum: 1 },
+      reasoning: { type: 'string', maxLength: 500 }
+    }
+  } as const
+
+  return {
+    async validateCommunityContent(request: ComplianceValidationRequest): Promise<ComplianceValidationResult> {
+      const startTime = Date.now()
+      const requestId = `${Date.now()}-${Math.random().toString(36).substr(2, 9)}`
+
+      try {
+        const { default: OpenAI } = await import('openai')
+        const openai = new OpenAI({ apiKey })
+
+        // Build user content
+        const userContent: any[] = [
+          {
+            type: 'text',
+            text: `Analyze this community content:\nName: "${request.name}"\nDescription: "${request.description}"\n\nReturn JSON matching the schema.`
+          }
+        ]
+
+        // Add image if provided
+        if (request.thumbnailBuffer) {
+          const dataUrl = `data:${request.thumbnailMime || 'image/png'};base64,${request.thumbnailBuffer.toString('base64')}`
+          userContent.push({ type: 'image_url', image_url: { url: dataUrl } })
+        }
+
+        // Production-optimized parameters
+        const completionParams = {
+          model,
+          // max_completion_tokens: 800, // Increased to allow reasoning + JSON output
+          response_format: {
+            type: 'json_schema' as const,
+            json_schema: {
+              name: 'ComplianceValidationResult',
+              schema: COMPLIANCE_SCHEMA
+            }
+          }
+        }
+
+        const messages = [
+          { role: 'system' as const, content: SYSTEM_PROMPT },
+          { role: 'user' as const, content: userContent }
+        ]
+
+        logger.info('Starting compliance validation', {
+          requestId,
+          name: request.name,
+          hasImage: String(!!request.thumbnailBuffer),
+          model
+        })
+
+        const res = await openai.chat.completions.create({
+          ...completionParams,
+          messages
+        })
+
+        // Production logging - only what's useful
+        if (res.usage) {
+          logger.info('OpenAI API usage', {
+            requestId,
+            promptTokens: res.usage.prompt_tokens,
+            completionTokens: res.usage.completion_tokens,
+            totalTokens: res.usage.total_tokens,
+            estimatedCost: `$${((res.usage.total_tokens / 1000) * 0.0001).toFixed(4)}` // GPT-5-nano pricing
+          })
+        }
+
+        const content = res.choices[0]?.message?.content
+        if (!content) {
+          logger.error('No content in OpenAI response', {
+            requestId,
+            choices: JSON.stringify(res.choices),
+            finishReason: res.choices[0]?.finish_reason,
+            response: JSON.stringify(res, null, 2)
+          })
+          throw new Error('No content received from OpenAI API')
```

Comment:
I am not sure if we can trust OpenAPI to return correct answers all the time, at least until we test the approach and maybe refine the prompt.

I am thinking about the need to react to Open API failures, so in case it cannot return a proper answer for a certain community, we leave that community on-hold and ask for a manual action (slack-message, maybe?). Having said so, I'd try to throw a specific error instance from this module so then we can react over it in a higher-level component. Basically:

`if (error instaceof ComplianceError) setCommunityAsPendingAndSendSlackMessage()`